### ISSUE TITLE
Separating Layers by Meshes and Materials w/o colorID

### ISSUE THREAD

**Issue by Ulazygit** [armory3d/armorpaint#537]
When importing an .obj you can either separate the model by meshes OR material. This means that when painting a layer, you can only filter out a mesh or a material respectively from the drop down menu.
![Capture](https://user-images.githubusercontent.com/60723284/90343162-d0c12a80-e005-11ea-9031-7b412cc2a316.JPG)

It would therefore be nice if we could have **two drop down menus and separate by material and mesh without having to go through the laborious process of baking a colorID map**. This would speed up workflow.

**Comment by AlexKiryanov** [comment]
Ulazygit Good day! I will correct! In the drop-down menu you can select an object or UV, but not any material. If I understand correctly what you are talking about.

**Comment by Ulazygit** [comment]
Hi Alex. I'm not sure we're on the same page here.

**Comment by AlexKiryanov** [comment]
Hello! Ulazigit! Sorry! I really misunderstood you. My English is not very good. I reread your post more carefully, and the campaign understood what you meant. Do you mean that after importing, you can divide the model, either into materials or objects with uv? like when you import a model? Not a bad idea.)

**Comment by Ulazygit** [comment]
Yes. Both at the same time. So something like this:
![mode](https://user-images.githubusercontent.com/60723284/90558220-9b4a4780-e193-11ea-9ecf-bcb0bf429a04.JPG)

**Comment by AlexKiryanov** [comment]
Yes that would be great! But I would not say that this will greatly shorten the time. Because if your model is divided into polygons with materials (I'm now talking about blender users), then bake the material ID map very quickly, maximum 5 minutes. And then import this map into the program and wipe it out using the ID tool just as quickly as choosing from the list. In general, the idea is not bad, just depending on how long it takes to remake the program for this option. Thanks for your reply.

**Comment by Ulazygit** [comment]
for me it takes too long to bake an ID map. Especially with 20+ materials

**Comment by AlexKiryanov** [comment]
try this addon, it will help bake the ID card in 1-2 minutes)
https://youtu.be/UbnHF4d6xPc

ср, 19 авг. 2020 г., 00:21 Ulazygit <notifications@github.com>:

**Comment by Ulazygit** [comment]
Thanks but I am still using 2.79

**Comment by AlexKiryanov** [comment]
Good morning! This plugin is also available for 2.79)

**Comment by luboslenco** [comment]
Keeping the existing workflow for now, but the color id map can be created quickly directly in armorpaint now. After importing the mesh and splitting it by material, use bake - object id to create the map.

<img width="1720" height="960" alt="Image" src="https://github.com/user-attachments/assets/517c72fe-1d0f-4842-b043-6066afe281c9" />
